deltachat-ios/AppDelegate.swift

```diff
@@ -111,6 +111,25 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UNUserNotificationCenterD
         logger.info("---- foreground ----")
         appIsInForeground = true
         startThreads()
+        let userDefaults = UserDefaults.init(suiteName: Constants.sharedUserDefaults)
+        if userDefaults?.bool(forKey: Constants.hasExtensionAttemptedToSend) ?? false {
+            let nc = NotificationCenter.default
+
+            DispatchQueue.main.async {
+                nc.post(
+                    name: dcNotificationChanged,
+                    object: nil,
+                    userInfo: [
+                        //TODO: Use correct message_id and chat_id
+                        "message_id": Int(0),
+                        "chat_id": Int(0),
+                        "date": Date(),
+                    ]
+                )
+            }
+
+            userDefaults?.removeObject(forKey: Constants.hasExtensionAttemptedToSend)
```

Comment:
maybe move this line before the notification is sent, as the first line after the "if": that would make race-conditions less likely (i know, they are already unlikely as the user cannot really do both), but, more important it keeps things closer together and improve readability.